### PR TITLE
Slash localization

### PR DESCRIPTION
**Pull request by Lulalaby**
TODO:

- [x] Check translation keys and ignore if not set
- [ ] Fix JSON examples

### CODE REVIEW

File: DisCatSharp.ApplicationCommands/Attributes/SlashCommand/OptionAttribute.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Linq;
+using DisCatSharp.Entities;
```
Comment (cking) [inline DisCatSharp.ApplicationCommands/Attributes/SlashCommand/OptionAttribute.cs:27]: needed?

File: DisCatSharp.ApplicationCommands/Attributes/SlashCommand/SlashCommandGroupAttribute.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@
 // SOFTWARE.
 
 using System;
+using DisCatSharp.Entities;
```
Comment (cking) [inline DisCatSharp.ApplicationCommands/Attributes/SlashCommand/SlashCommandGroupAttribute.cs:24]: needed?

File: DisCatSharp/Entities/Application/DiscordApplicationCommandLocalization.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        /// <summary>
+        /// Gets the localization dict.
+        /// </summary>
+        public Dictionary<string, string> Localizations { get; internal set; }
+
+        /// <summary>
+        /// Gets valid locales for Discord.
+        /// </summary>
+        internal List<string> _validLocales = new() { "ru", "fi", "hr", "de", "hu", "sv-SE", "cs", "fr", "it", "en-GB", "pt-BR", "ja", "tr", "en-US", "es-ES", "uk", "hi", "th", "el", "no", "ro", "ko", "zh-TW", "vi", "zh-CN", "pl", "bg", "da", "nl", "lt" };
```
Comment (cking) [inline DisCatSharp/Entities/Application/DiscordApplicationCommandLocalization.cs:43]: is it really a fixed, predetermined list? just wondering

**Review by cking — DISMISSED** [review]

File: DisCatSharp/Entities/Application/DiscordApplicationCommandLocalization.cs (hunk 3 above)
Comment (Lulalaby) [reply DisCatSharp/Entities/Application/DiscordApplicationCommandLocalization.cs:43]: Yes. Discord uses its own system.
The API only accept these.
See https://docs.dcs.aitsys.dev/localization/articles/application_commands/translations/reference.html#valid-locales

File: DisCatSharp.ApplicationCommands/Attributes/SlashCommand/OptionAttribute.cs (hunk 1 above)
Comment (Lulalaby) [reply DisCatSharp.ApplicationCommands/Attributes/SlashCommand/OptionAttribute.cs:27]: prob not now

File: DisCatSharp.ApplicationCommands/Attributes/SlashCommand/SlashCommandGroupAttribute.cs (hunk 2 above)
Comment (Lulalaby) [reply DisCatSharp.ApplicationCommands/Attributes/SlashCommand/SlashCommandGroupAttribute.cs:24]: prob not now

**Review by cking — DISMISSED** [review]

**Pull request merged**